.github/workflows/ci.yml

```
@@ -0,0 +1,51 @@
+name: CI
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+
+# Cancel active CI runs for a PR before starting another run
+concurrency:
+  group: ${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  pre-commit:
+    name: Pre-commit checks
+    uses: beeware/.github/.github/workflows/pre-commit-run.yml@main
```

I missed that it is not some generic pre-commit action.  I think we should just find/use generic precommit thingie here. Pasting from https://github.com/pre-commit/action

```suggestion
    name: Pre-commit checks
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v3
    - uses: actions/setup-python@v3
    - uses: pre-commit/action@v3.0.0
```